geomstats/stratified_geometry/spider.py

```diff
@@ -0,0 +1,303 @@
+            return False
+        return True
+
+    @staticmethod
+    def _value_check(single_point):
+        r"""Check if a random point has the correct length.
+
+        Parameters
+        ----------
+        single_point : SpiderPoint
+             Point to be checked.
+
+        Returns
+        -------
+        belongs : boolean
+            Boolean denoting if the point has a positive length when on non-zero ray.
+        """
+        if single_point.s != 0 and single_point.x <= 0:
+            return False
+        return True
+
+    @_vectorize_point((1, "point"))
+    def set_to_array(self, point):
+        r"""Turn a point into an array compatible with the dimension of the space.
+
+        Parameters
+        ----------
+        point : SpiderPoint or list of SpiderPoint, shape=[...]
+             Points to be checked.
+
+        Returns
+        -------
+        point_array : array-like, shape=[...,rays]
+            An array with the x parameter in the s position.
+        """
+        point_to_array = gs.zeros((len(point), self.rays))
+        for i, pt in enumerate(point):
+            point_to_array[i, pt.s - 1] = pt.x
+        return point_to_array
+
+
+class SpiderGeometry(PointSetGeometry):
+    """Geometry on the Spider, induced by the rays Geometry."""
+
+    def __init__(self, space, ambient_metric=EuclideanMetric(1)):
+        super(SpiderGeometry, self).__init__(space=space)
+        self.rays_geometry = ambient_metric
+
+    @property
+    def rays(self):
+        """Get rays."""
+        return self.space.rays
+
+    @_vectorize_point((1, "a"), (2, "b"))
+    def dist(self, point_a, point_b):
+        """Compute the distance between two points on the Spider using the ray geometry.
+
+        Parameters
+        ----------
+        point_a : SpiderPoint or list of SpiderPoint, shape=[...]
+             Point in the Spider.
+        point_b : SpiderPoint or list of SpiderPoint, shape=[...]
+             Point in the Spider.
+
+        Returns
+        -------
+        point_array : array-like, shape=[...]
+            An array with the distance.
+        """
+        if len(point_a) == 1:
+            values = itertools.zip_longest(point_a, point_b, fillvalue=point_a[0])
+        elif len(point_b) == 1:
+            values = itertools.zip_longest(point_a, point_b, fillvalue=point_b[0])
+        else:
+            values = itertools.zip_longest(point_a, point_b)
+
+        result = []
+        for point_a_, point_b_ in values:
+            if point_a_.s == point_b_.s or point_a_.s == 0 or point_b_.s == 0:
+                result += [self.rays_geometry.norm(gs.array([point_a_.x - point_b_.x]))]
+            else:
+                result += [point_a_.x + point_b_.x]
+        return gs.array(result) if len(result) != 1 else result[0]
+
+    @_vectorize_point((1, "initial_point"), (2, "end_point"))
+    def geodesic(self, initial_point, end_point):
+        """Return the geodesic between two lists of Spider points.
```

Comment:
Why between two lists of spider points?

I thought:
- the space is the spider
- one point in this space is a spiderpoint
- a curve on that space is a curve between 2 spiderpoints

Are we infact implicitly considering a product space of N spiderspaces? In this case, this is very linked to the ProductManifold and NFoldManifold that we have. 

We can leave this for further discussion, but it is a bit confusing to me to unify the space of SpiderPoints and the space of N SpiderPoints.